cassandra/protocol.py

```diff
@@ -1015,6 +994,65 @@ def decode_message(cls, protocol_version, user_type_map, stream_id, flags, opcod
         return msg
 
 
+def cython_protocol_handler(colparser):
+    """
+    Given a column parser to deserialize ResultMessages, return a suitable
+    Cython-based protocol handler.
+
+    There are three Cython-based protocol handlers (least to most performant):
+
+        1. objparser.ListParser
+            this parser decodes result messages into a list of tuples
+
+        2. objparser.LazyParser
+            this parser decodes result messages lazily by returning an iterator
+
+        3. numpyparser.NumPyParser
+            this parser decodes result messages into NumPy arrays
+
+    The default is to use objparser.ListParser
+    """
+    # TODO: It may be cleaner to turn ProtocolHandler and ResultMessage into
+    # TODO:     instances and use methods instead of class methods
```

Comment:
I see. Either way works for me, do whatever feels best, as the closure
itself isn't a problem. I'll remove the leftover TODOs.

On Tue, Aug 11, 2015 at 9:32 PM Adam Holmberg <notifications@github.com>
wrote:

> In cassandra/protocol.py
> <https://github.com/datastax/python-driver/pull/382#discussion_r36794804>:
>
> > +
> > +    There are three Cython-based protocol handlers (least to most performant):
> > +
> > +        1. objparser.ListParser
> > +            this parser decodes result messages into a list of tuples
> > +
> > +        2. objparser.LazyParser
> > +            this parser decodes result messages lazily by returning an iterator
> > +
> > +        3. numpyparser.NumPyParser
> > +            this parser decodes result messages into NumPy arrays
> > +
> > +    The default is to use objparser.ListParser
> > +    """
> > +    # TODO: It may be cleaner to turn ProtocolHandler and ResultMessage into
> > +    # TODO:     instances and use methods instead of class methods
>
> Well, it's exposed to the degree that it's specified in this config
> parameter
> <https://github.com/markflorisson/python-driver/blob/9240c71c02c27a6f20c73870a92ccdafab238df7/cassandra/cluster.py#L1524>.
> This pattern was chosen for simplicity there, and to match the
> connection_class parameter, which supports a similar pluggable approach
> using class types.
> I could be convinced either way.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/datastax/python-driver/pull/382/files#r36794804>.
>